### PR TITLE
Upgrade to v2.15.0 of the Docker Terraform Provider

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #285
Fixes: #287
Fixes: #288
Fixes: #295
Fixes: #297
Fixes: #305﻿

### CODE REVIEW

File: provider/resources.go [hunk 1]
```diff
@@ -115,6 +114,7 @@ func Provider() tfbridge.ProviderInfo {
 			"docker_network":        {Tok: dockerDataSource(dockerMod, "getNetwork")},
 			"docker_registry_image": {Tok: dockerDataSource(dockerMod, "getRegistryImage")},
 			"docker_plugin":         {Tok: dockerDataSource(dockerMod, "getPlugin")},
+			"docker_image":          {Tok: dockerDataSource(dockerMod, "getImage")},
```
Comment (mikhailshilkov) [inline provider/resources.go:117]: Should it be called `getRemoteImage`?

File: sdk/dotnet/Config/Config.cs [hunk 2]
```diff
@@ -33,7 +33,7 @@ public static class Config
         /// </summary>
         public static string? KeyMaterial { get; set; } = __config.Get("keyMaterial");
 
-        public static ImmutableArray<Pulumi.Docker.Config.Types.RegistryAuth> RegistryAuth { get; set; } = __config.GetObject<ImmutableArray<Pulumi.Docker.Config.Types.RegistryAuth>>("registryAuth");
+        public static Pulumi.Docker.Config.Types.RegistryAuth? RegistryAuth { get; set; } = __config.GetObject<Pulumi.Docker.Config.Types.RegistryAuth>("registryAuth");
```
Comment (mikhailshilkov) [inline sdk/dotnet/Config/Config.cs:36]: This looks breaking... do you know what happened here?

**Review by mikhailshilkov — COMMENTED** [review]
Do we also need to changelog the breaking changes?

File: sdk/dotnet/Config/Config.cs (hunk 2 above)
Comment (stack72) [reply sdk/dotnet/Config/Config.cs]: Fixed! They added MAxItems: 1, I removed it again to ensure we don't break - it was a linting change on their side

**Review by mikhailshilkov — APPROVED** [review]

**Pull request merged**